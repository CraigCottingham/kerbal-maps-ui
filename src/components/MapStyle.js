import React from "react"

class MapStyle extends React.Component {
  constructor(props) {
    super(props)
    this.changeValue = this.changeValue.bind(this)
    this.options = this.props.options
  }

  changeValue(event) {
    this.props.onValueChange(event.target.value)
  }

  render() {
    const selectedValue = this.props.selectedValue
    const options = this.options.map((option) => <option key={"style_" + option.value} value={option.value}>{option.label}</option>)

    return (
      <div className="form-group">
        <label htmlFor="select-map-style">Style</label>
        <select id="select-map-style" name="select-map-style" value={selectedValue} className="form-control" onChange={this.changeValue}>
          {options}
        </select>
      </div>
    )
  }
}

export default MapStyle
