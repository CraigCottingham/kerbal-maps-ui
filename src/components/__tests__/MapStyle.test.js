import React from "react"
import MapStyle from "../MapStyle"
import renderer from "react-test-renderer"

it("renders correctly", () => {
  const options = [
    {value: "a", label: "A"},
    {value: "b", label: "B"},
    {value: "c", label: "C"}
  ]

  const tree = renderer
    .create(
      <MapStyle options={options} selectedValue={"a"} />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
