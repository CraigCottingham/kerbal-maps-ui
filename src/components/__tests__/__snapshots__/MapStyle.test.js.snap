// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="select-map-style"
  >
    Style
  </label>
  <select
    className="form-control"
    id="select-map-style"
    name="select-map-style"
    onChange={[Function]}
    value="a"
  >
    <option
      value="a"
    >
      A
    </option>
    <option
      value="b"
    >
      B
    </option>
    <option
      value="c"
    >
      C
    </option>
  </select>
</div>
`;
